src/citrine/informatics/predictors.py

```diff
@@ -272,3 +274,165 @@ def __str__(self):
+    id_to_quantity = _properties.Mapping(_properties.String, _properties.Object(RealDescriptor),
+                                         'config.id_to_quantity')
+    labels = _properties.Mapping(_properties.String, _properties.List(_properties.String),
+                                 'config.labels')
+    typ = _properties.String('config.type', default='IngredientsToSimpleMixture',
+                             deserializable=False)
+    status = _properties.Optional(_properties.String(), 'status', serializable=False)
+    status_info = _properties.Optional(
+        _properties.List(_properties.String()),
+        'status_info',
+        serializable=False
+    )
+    active = _properties.Boolean('active', default=True)
+
+    # NOTE: These could go here or in _post_dump - it's unclear which is better right now
+    module_type = _properties.String('module_type', default='PREDICTOR')
+    schema_id = _properties.UUID('schema_id', default=UUID('873e4541-da8a-4698-a981-732c0c729c3d'))
+
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 output: FormulationDescriptor,
+                 id_to_quantity: Dict[str, RealDescriptor],
+                 labels: Dict[str, List[str]],
+                 session: Optional[Session] = None,
+                 report: Optional[Report] = None,
+                 active: bool = True):
+        self.name: str = name
+        self.description: str = description
+        self.output: FormulationDescriptor = output
+        self.id_to_quantity: Dict[str, RealDescriptor] = id_to_quantity
+        self.labels: Dict[str, List[str]] = labels
+        self.session: Optional[Session] = session
+        self.report: Optional[Report] = report
+        self.active: bool = active
+
+    def _post_dump(self, data: dict) -> dict:
+        data['display_name'] = data['config']['name']
+        return data
+
+    def __str__(self):
+        return '<IngredientsToSimpleMixturePredictor {!r}>'.format(self.name)
+
+    def post_build(self, project_id: UUID, data: dict):
+        """Creates the predictor report object."""
+        self.report = ReportResource(project_id, self.session).get(data['id'])
+
+
+class GeneralizedMeanPropertyPredictor(
+        Serializable['GeneralizedMeanPropertyPredictor'], Predictor):
+    """[ALPHA] A predictor interface that mean component properties.
+
+    Parameters
+    ----------
+    name: str
+        name of the configuration
+    description: str
+        description of the predictor
+    input_descriptor: FormulationDescriptor
+        descriptor that represents the input formulation
+    properties: List[str]
+        List of component properties to featurize
+    p: float
+        Power of the generalized mean
+    impute_properties: bool
+        Whether to impute missing component properties
+    training_data: DataSource
+        Source of the training data, which can be either a CSV or an Ara table
+    label: Optional[str]
```

Comment:
If specified, only ingredients with the label are featurized. Otherwise, all ingredients in the mixture featurized.

For example, we may wish to featurize the average solvent density in a mixture instead of the average density of all ingredients.